tests/unit/backend/corpora/api_server/curator/collection/test_curation_collections.py

```diff
@@ -409,16 +410,118 @@ def test__update_collection__OK(self):
+
+    def test__update_collection__links_and_doi_management__OK(self):
+        name = "partial updates test collection"
+        links = [
+            {"link_name": "name", "link_type": "RAW_DATA", "link_url": "http://test_link.place"},
+            {"link_name": "doi", "link_type": "DOI", "link_url": "http://doi.doi"},
+        ]
+        new_links = [
+            {"link_name": "new link", "link_type": "RAW_DATA", "link_url": "http://brand_new_link.place"},
+            {"link_name": "new doi", "link_type": "DOI", "link_url": "http://doi.org/10.1016"},  # a real DOI
+        ]
+        new_links_no_doi = [
+            {"link_name": "new link", "link_type": "RAW_DATA", "link_url": "http://brand_new_link.place"},
+        ]
+
+        links_configurations = (
+            ("With links already in place; new links replace old", links, new_links, 200, new_links),
+            ("With no links in place; new links get added", None, new_links, 200, new_links),
+            ("With links in place, but empty request; no changes are made", links, None, 200, links),
+            ("With links in place, empty array passed; BAD REQUEST 400", links, [], 400, links),
+            ("With links but no doi; doi remains in place", links, new_links_no_doi, 200, new_links_no_doi + links[1:]),
+        )
+
+        for test_title, initial_links, new_links, expected_status_code, expected_links in links_configurations:
+            with self.subTest(test_title):
+                collection_id = self.generate_collection(self.session, name=name, links=initial_links).id
+                original_collection = self.app.get(f"curation/v1/collections/{collection_id}").json
+                self.assertEqual(initial_links if initial_links else [], original_collection["links"])
+                metadata = {"links": new_links} if new_links is not None else {}
+                response = self.app.patch(
+                    f"/curation/v1/collections/{collection_id}",
+                    data=json.dumps(metadata),
+                    headers=self.make_owner_header(),
+                )
+                self.assertEqual(expected_status_code, response.status_code)
+                if expected_status_code == 200:
+                    self.assertEqual(name, response.json["name"])
+                    self.assertEqual(expected_links, response.json["links"])
+
+    def test__update_collection__doi_does_not_exist__BAD_REQUEST(self):
+        name = "bad doi update test collection"
+        links = [
+            {"link_name": "name", "link_type": "RAW_DATA", "link_url": "http://test_link.place"},
+            {"link_name": "doi", "link_type": "DOI", "link_url": "http://doi.doi"},
+        ]
+        new_links_bad_doi = [
+            {"link_name": "new link", "link_type": "RAW_DATA", "link_url": "http://brand_new_link.place"},
+            {"link_name": "new doi", "link_type": "DOI", "link_url": "http://a_bad_doi"},  # a bad DOI
```

Comment:
Make sure the request to the DOI sever is mocked out.

Reply:
It doesn't need to be. It is handled by the existing code. The request results in a return value of `None` for the new doi; this is handled correctly by the existing algorithm in the patch request handler.

Reply:
It is handled below by the `with patch(...)` code on ln 500